graphqlapi/dynamic_generation_converted_fetch.go

```diff
@@ -0,0 +1,312 @@
+/*                          _       _
+ *__      _____  __ ___   ___  __ _| |_ ___
+ *\ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+ * \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+ *  \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+ *
+ * Copyright © 2016 - 2018 Weaviate. All rights reserved.
+ * LICENSE: https://github.com/creativesoftwarefdn/weaviate/blob/develop/LICENSE.md
+ * AUTHOR: Bob van Luijt (bob@kub.design)
+ * See www.creativesoftwarefdn.org for details
+ * Contact: @CreativeSofwFdn / bob@kub.design
+ */
+
+package graphqlapi
+
+import (
+	"fmt"
+	"github.com/creativesoftwarefdn/weaviate/models"
+	"github.com/creativesoftwarefdn/weaviate/schema"
+	"github.com/graphql-go/graphql"
+)
+
+// Build the dynamically generated ConvertedFetch Actions part of the schema
+func (g *GraphQL) genActionClassFieldsFromSchema(convertedFetchActionsAndThings *map[string]*graphql.Object) (*graphql.Object, error) {
+
+	actionClassFields := graphql.Fields{}
+
+	for _, class := range g.databaseSchema.ActionSchema.Schema.Classes {
+
+		singleActionClassField, singleActionClassObject, err := genSingleActionClassField(class, convertedFetchActionsAndThings)
+
+		if err != nil {
+			return nil, err
+		}
+		actionClassFields[class.Class] = singleActionClassField
+
+		(*convertedFetchActionsAndThings)[class.Class] = singleActionClassObject
+	}
+
+	localConvertedFetchActions := graphql.ObjectConfig{
+		Name:        "WeaviateLocalConvertedFetchActionsObj",
+		Fields:      actionClassFields,
+		Description: "Fetch Actions on the internal Weaviate",
+	}
+
+	return graphql.NewObject(localConvertedFetchActions), nil
+}
+
+func genSingleActionClassField(class *models.SemanticSchemaClass, convertedFetchActionsAndThings *map[string]*graphql.Object) (*graphql.Field, *graphql.Object, error) {
+	singleActionClassPropertyFields := graphql.ObjectConfig{
+		Name: class.Class,
+		Fields: (graphql.FieldsThunk)(func() graphql.Fields {
+			singleActionClassPropertyFields, err := genSingleActionClassPropertyFields(class, convertedFetchActionsAndThings)
+			if err != nil {
+				panic("oops")
```

Comment:
Agreed, but I'm afraid this is currently impossible because we have to cast Fields as a FieldsThunk to handle cyclical relationships between Classes. A FieldsThunk requires a func() that returns Fields. There is no way to return an additional (error) value here.